common/pango.c

```diff
@@ -88,14 +89,22 @@ PangoLayout *get_pango_layout(cairo_t *cairo, const char *font,
 
 void get_text_size(cairo_t *cairo, const char *font, int *width, int *height,
 		int *baseline, double scale, bool markup, const char *fmt, ...) {
-	char buf[MAX_CHARS];
-
 	va_list args;
 	va_start(args, fmt);
-	if (vsnprintf(buf, sizeof(buf), fmt, args) >= MAX_CHARS) {
-		strcpy(&buf[sizeof(buf) - sizeof(overflow)], overflow);
-	}
+	// Add one since vsnprintf excludes null terminator.
+	int length = vsnprintf(NULL, 0, fmt, args) + 1;
 	va_end(args);
+	// Limit length of string to MAX_CHARS.
+	int length_limited = imin(length, MAX_CHARS);
```

Comment:
TBH I don't think limiting to `MAX_CHARS` is necessary.

Reply:
Unlimited size? I guess the only benefit is protection against oversized strings created from a bug or something like that.

Reply:
Yeah, we don't do this elsewhere, let's not do it here.